### PR TITLE
cli: `delete --all`

### LINKED ISSUE #304: cli: testsys delete --all
It would be nice to have a command to delete all testsys tests/resources/controllers and the namespace from a cluster.

### PR DESCRIPTION
**Pull request by ecpullen**
**Issue number:**

Closes #304 

**Description of changes:**

Adds a `--all` flag to the delete command to delete all testsys object in a _smart_ way. First, all tests are deleted at the same time. After the tests are deleted resources are deleted in batches based on their dependencies. This allows us to delete a set of many tests in a similar amount of time as a single test.

**Testing done:**

Ran `testsys run aws-ecs --upgrade-downgrade` 2 times and used `testsys delete --all` to clean everything up.

Output:
```
~/bottlerocket-test-system% testsys delete --all
Deleting all tests...
Test deletion complete.
Deleting all resources following dependencies...
Deleting resource 'testsys-test2-instances' ...
Deleting resource 'testsys-test1-instances' ...
Resource 'testsys-test1-instances' has been deleted
Resource 'testsys-test2-instances' has been deleted
Deleting resource 'testsys-test1' ...
Deleting resource 'testsys-test2' ...
Resource 'testsys-test1' has been deleted
Resource 'testsys-test2' has been deleted
All resources have been deleted.
```

Help:
```
~/bottlerocket-test-system% testsys delete -h
testsys-delete 0.1.0
Delete a testsys object

USAGE:
    testsys delete [FLAGS] <object-type> <object-name>

FLAGS:
        --all                  Delete all tests and resources from a testsys cluster
    -h, --help                 Prints help information
        --include-resources    Include this flag if all resources this object depends on should be deleted as well
    -V, --version              Prints version information

ARGS:
    <object-type>    The type of object that is being delete must be either `test` or `resource`
    <object-name>    The name of the test/resource that should be deleted
```

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Comment by ecpullen** [comment]
^ clippy fixes

File: testsys/src/delete.rs [hunk 1]
```diff
@@ -36,15 +43,81 @@ derive_fromstr_from_deserialize!(ObjectType);
...
+        what: "all tests".to_string(),
+    })?;
+    while !test_client
+        .get_all()
+        .await
+        .context(error::GetSnafu { what: "all tests" })?
+        .is_empty()
+    {
+        tokio::time::sleep(Duration::from_secs(5)).await;
+    }
```
Comment (webern) [inline testsys/src/delete.rs:78]: We should move the `wait` procedures out and call them with timeouts. I'm not sure how long to wait, a few minutes?

Comment (ecpullen) [reply testsys/src/delete.rs:78]: I am worried about setting timeouts because deleting a cluster takes forever.

Comment (webern) [reply testsys/src/delete.rs:78]: Makes sense.

**Review by webern — APPROVED** [review]

**Review by etungsten — APPROVED** [review]

**Pull request merged**